src/erlcloud_aws.erl

```diff
@@ -293,6 +290,15 @@ format_timestamp({{Yr, Mo, Da}, {H, M, S}}) ->
       io_lib:format("~4.10.0b-~2.10.0b-~2.10.0bT~2.10.0b:~2.10.0b:~2.10.0bZ",
                     [Yr, Mo, Da, H, M, S])).
 
+encode_params(Params, Headers) ->
+  LowerCaseHeaders = lists:map(fun({K, V}) -> {string:to_lower(K), V} end, Headers),
+  case proplists:get_value(<<"content-type">>, LowerCaseHeaders) of
+    undefined -> {erlcloud_http:make_query_string(Params),
+                  [{"content-type", ?DEFAULT_CONTENT_TYPE} | LowerCaseHeaders]};
+    % when `content-type` is specified, this is developer's responsibility to encode params properly
```

Comment:
seems to be part of readme or at least top level `aws_request` fun

Reply:
moved it on the `aws_request4/9` level, because this is the only place where user can provide headers (including content type) and params that need to be encoded.
I've also added the content type check to the `aws_request_form` (previously it was the place where headers were provisioned with default content type) because this function is also exposed